File: src/devtwin/services/postgres.py
"""PostgreSQL service detector."""

from __future__ import annotations

from devtwin.services.base import ServiceDetector


class PostgresDetector(ServiceDetector):
    name = "postgresql"
    default_port = 5432
    env_var_patterns = (r"^POSTGRES", r"^PG(HOST|PORT|USER|PASSWORD|DATABASE)$", r"^DATABASE_URL$")
    dependency_names = ("psycopg2", "psycopg2-binary", "psycopg", "pg", "asyncpg", "postgres")
    compose_image_hints = ("postgres", "postgresql", "timescale")
    readme_keywords = ("postgres", "postgresql")

File: mcphub/devtwin/src/services/postgres.ts
/** PostgreSQL service detector. */

import { ServiceDetector } from './base.js';

export class PostgresDetector extends ServiceDetector {
  readonly name = 'postgresql';
  readonly defaultPort = 5432;
  override readonly envVarPatterns = [
    '^POSTGRES',
    '^PG(HOST|PORT|USER|PASSWORD|DATABASE)$',
    '^DATABASE_URL$',
  ];
  override readonly dependencyNames = [
    'psycopg2',
    'psycopg2-binary',
    'psycopg',
    'pg',
    'asyncpg',
    'postgres',
  ];
  override readonly composeImageHints = ['postgres', 'postgresql', 'timescale'];
  override readonly readmeKeywords = ['postgres', 'postgresql'];
}
